OloEngine/src/OloEngine/Renderer/GPUResourceQueue.cpp

```
@@ -0,0 +1,227 @@
+#include "OloEnginePCH.h"
+#include "GPUResourceQueue.h"
+#include "OloEngine/Renderer/Texture.h"
+#include "OloEngine/Renderer/Shader.h"
+
+namespace OloEngine
+{
+    // Static member definitions
+    std::queue<std::unique_ptr<GPUResourceCommand>> GPUResourceQueue::s_CommandQueue;
+    std::mutex GPUResourceQueue::s_QueueMutex;
```

Clang (14.0.6)

[warning] variable 's_QueueMutex' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)